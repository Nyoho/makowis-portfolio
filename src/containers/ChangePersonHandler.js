// @flow
/**
 * Personの変化に検知して動作する処理をまとめたコンポーネント
 * 各コンポーネントに対して不要な依存関係を持ち込まないために使う
 */
import React from 'react';
import { connect } from 'react-redux';
import type { Dispatch } from 'redux';

import type { Person } from '../models/Person';
import getSlides from '../api/getSlides';
import type { State } from '../reducers/index';

const changeSlides = (dispatch: Dispatch) => async ({
  speakerdeck,
}: Person) => {
  // 一旦表示してるスライドを消す
  dispatch({ type: 'CHANGE_SLIDES', slides: [] });
  if (speakerdeck) {
    const slides = await getSlides(speakerdeck);
    dispatch({ type: 'CHANGE_SLIDES', slides });
  }
};

type Props = {
  person: Person,
  onChangePerson: (Person) => Promise<void>,
};

export const Component = ({ onChangePerson, person }: Props) => {
  onChangePerson(person);
  return <div style={{ display: 'none' }} />;
};

const mapStateToProps = ({ person }: State) => ({ person });
const mapDispatchToProps = (dispatch: Dispatch) => ({
  onChangePerson: changeSlides(dispatch),
});

const ChangePersonHandler = connect(mapStateToProps, mapDispatchToProps)(
  Component,
);

export default ChangePersonHandler;
