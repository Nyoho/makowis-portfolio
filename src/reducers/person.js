// @flow
import person from '../models/Person';
import type { Person } from '../models/Person';

const CHANGE_PERSON: 'CHANGE_PERSON' = 'CHANGE_PERSON';

export type State = Person;

type Action = { type: typeof CHANGE_PERSON, person: Person };

const reducer = (state: Person = person, action: Action): State => {
  switch (action.type) {
    case CHANGE_PERSON:
      return action.person;
    default:
      return state;
  }
};

export default reducer;
