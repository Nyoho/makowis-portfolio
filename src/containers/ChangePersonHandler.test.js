/* eslint-env jest */
// @flow
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Component } from './ChangePersonHandler';
import person from '../models/Person';

const reducer = () => ({ person: {}, slides: [] });

const store = createStore(reducer);

it('renders Person Developers', () => {
  const tree = renderer
    .create(
      <Provider {...{ store }}>
        <Component
          onChangePerson={() => Promise.resolve(undefined)}
          person={person}
        />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
