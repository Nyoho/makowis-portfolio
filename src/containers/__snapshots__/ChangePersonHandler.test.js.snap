// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Person Developers 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
/>
`;
